In [ ]:
#1. Explain the dataset

In [14]:
import pandas as pd
add_media = pd.read_csv('Advertising.csv')
add_media

Unnamed: 0     TV  Radio  Newspaper  Sales
0             1  230.1   37.8       69.2   22.1
1             2   44.5   39.3       45.1   10.4
2             3   17.2   45.9       69.3    9.3
3             4  151.5   41.3       58.5   18.5
4             5  180.8   10.8       58.4   12.9
..          ...    ...    ...        ...    ...
195         196   38.2    3.7       13.8    7.6
196         197   94.2    4.9        8.1    9.7
197         198  177.0    9.3        6.4   12.8
198         199  283.6   42.0       66.2   25.5
199         200  232.1    8.6        8.7   13.4

[200 rows x 5 columns]

In [15]:
add_media.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  200 non-null    int64  
 1   TV          200 non-null    float64
 2   Radio       200 non-null    float64
 3   Newspaper   200 non-null    float64
 4   Sales       200 non-null    float64
dtypes: float64(4), int64(1)
memory usage: 7.9 KB


In [6]:
#2. Check the summary statistics and discuss the max, min, avg, median,and percentiles

In [16]:
add_media.describe(exclude=[int])

TV       Radio   Newspaper       Sales
count  200.000000  200.000000  200.000000  200.000000
mean   147.042500   23.264000   30.554000   14.022500
std     85.854236   14.846809   21.778621    5.217457
min      0.700000    0.000000    0.300000    1.600000
25%     74.375000    9.975000   12.750000   10.375000
50%    149.750000   22.900000   25.750000   12.900000
75%    218.825000   36.525000   45.100000   17.400000
max    296.400000   49.600000  114.000000   27.000000

2.1 Analysing Maximum value 

Advertising through TV       :296.40 
                   Radio     :49.60 
                   Newspaper :114.00 
                   sales     :27.00
Hightest spent on TV advertising

2.2 Analysing Minimum value 
                    TV       :0.7
                   Radio     :0
                   Newspaper :0.3 
                   sales     :1.6                 

Lowest spent on Radio

2.3 Analysing Average/Mean 
                   TV       :147.04
                   Radio     :23.26
                   Newspaper :30.55
                   sales     :14.02   
Mean advertisement expenditure across columns,Tv occupies first position, Newspaper occupies second position and Radio third position.

2.4 Analysing median value across columns
                   TV       :149.75
                   Radio     :22.9
                   Newspaper :25.75
                   sales     :12.9
 Comparing median advertisement expenditure across columns,Tv occupies first position, Newspaper occupies second position and Radio third position.


2.5 Analysing percentiles across columns

Percentiles(TV column):

25% of the values are less than or equal to 74.375 OR 75 % of the values are greater than or equal to 74.375; 50 % of the values are less than or equal to 149.75 OR 50 % of the values are greater than or equal to 149.75 (the 50 percentile defines median);75% of the values are less than or equal to 218.825 OR 25 % of the values are greater than or equal to 218.825

Percentiles(Radio column):

25% of the values are less than or equal to 9.975;50 % of the values are less than or equal to 22.90 ;75% of the values are less than or equal to 36.525

Percentiles(Newspaper column):

25% of the values are less than or equal to 12.75;50 % of the values are less than or equal to 25.75 ;75% of the values are less than or equal to 45.10



In [ ]:
#3. The manager wants to find out whether the same amount was spent for the three advertisements (TV , Radio and Newspaper). Comment
#on your findings.

In [ ]:
# Stating the hypothesis


H0= Same amount was spent for the three advertisements,µ1=µ2=µ3

Ha= Amount spent for atleast one advertisement differs from the rest,All μ are not equal

In [ ]:
# setting the significance level

Considering significance level(α)=0.05 or 5%

In [ ]:
# Determining p value using one way anova (for comparing means from 3 groups)

In [17]:
import numpy as np
from scipy.stats import f_oneway
x=np.array(add_media['TV'])
y=np.array(add_media['Radio'])
z=np.array(add_media['Newspaper'])
f_oneway(x,y,z)

F_onewayResult(statistic=358.8514595342597, pvalue=4.552931539744962e-103)

In [ ]:
# comparing p value with significance level(0.05) 

p value: 4.552931539744962e-103  < 0.05
Rejecting the null hypothesis

Since p value is less than the selected significance level of 0.05 ,rejecting the null hypothesis
The amount spent for atleast one advertisement differs from the rest i.e, All μ are not equal